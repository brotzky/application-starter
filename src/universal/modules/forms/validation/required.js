export default function(value) {
  // Allow 0
  if (typeof value === 'string') {
    if (value.trim().length <= 0) {
      return {
        defaultMessage: ' is required',
        id: 'form.errors.isRequired',
        values: {},
      };
    }
  }
  return value !== undefined && value !== null && value !== ''
    ? undefined
    : {
        defaultMessage: ' is required',
        id: 'form.errors.isRequired',
        values: {},
      };
}
