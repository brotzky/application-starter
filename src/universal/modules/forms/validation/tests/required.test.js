import required from '../required';

describe('[Form Validation] required', () => {
  describe('required()', () => {
    it('should return an error object if nothing is passed it', () => {
      expect(required()).toMatchSnapshot();
    });

    it('should return no errors if a string is passed in', () => {
      expect(required('string')).toMatchSnapshot();
    });

    it('should return no errors if 0 is passed in', () => {
      expect(required(0)).toMatchSnapshot();
    });
  });
});
