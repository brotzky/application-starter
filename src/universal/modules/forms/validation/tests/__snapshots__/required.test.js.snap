// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] required required() should return an error object if nothing is passed it 1`] = `
Object {
  "defaultMessage": " is required",
  "id": "form.errors.isRequired",
  "values": Object {},
}
`;

exports[`[Form Validation] required required() should return no errors if 0 is passed in 1`] = `undefined`;

exports[`[Form Validation] required required() should return no errors if a string is passed in 1`] = `undefined`;
